CLDR-18544 Add transforms for translteration Adlam <-> Latin

sven-oly opened the pull request:
CLDR-18544

- [x] This PR completes the ticket.

<!--
Thank you for your pull request.
Please see https://cldr.unicode.org/index/process for general
information on contributing to CLDR.

1. Make sure the ticket is filed at
https://unicode-org.atlassian.net/projects/CLDR/
2. Update the PR title and first line of this
message to include the ticket ID (CLDR-18544)
3. You will be automatically asked to sign the contributors’
license before the PR is accepted.
- sign: https://cla-assistant.io/unicode-org/cldr
- license: https://www.unicode.org/copyright.html#License
-->

ALLOW_MANY_COMMITS=true

macchiati reviewed (changes requested):
Test files go into https://github.com/unicode-org/cldr/tree/main/common/testData/transforms and must follow the structure in those files. 
- must be named <transformId>.txt
- lines must be in the format: <source text><TAB><expected result text>
- should be extensive, eg 
    - https://github.com/unicode-org/cldr/blob/main/common/testData/transforms/und-Arab-t-und-beng.txt
    - https://github.com/unicode-org/cldr/blob/main/common/testData/transforms/cs-fonipa-t-cs.txt

sven-oly commented:
Thanks for the notes on this PR.

I've moved test files to the right directory and added test strings including region names, days, months, some geographic features, and a few other values.
